### PR TITLE
AEA-1302 Add data dir usage to p2p_libp2p and stub connections

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR fixes some of the connections to use the correct directory for data.

## Fixes

- #2178 

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: packages/fetchai/connections/p2p_libp2p/connection.py [hunk 1]
```diff
@@ -202,11 +210,12 @@ def __init__(
 
         # log file
         self.log_file = log_file if log_file is not None else LIBP2P_NODE_LOG_FILE
-        self.log_file = os.path.join(os.path.abspath(os.getcwd()), self.log_file)
-
+        if not Path(self.log_file).is_absolute():
+            self.log_file = os.path.abspath(os.path.join(data_dir, self.log_file))
```
Comment (marcofavorito) [inline packages/fetchai/connections/p2p_libp2p/connection.py:214]: @DavidMinarsch I noticed this:

```python
import os
p1 = "/some/abs/path
p2 = "/another/abs/path"
p3 = os.path.join(p1, p2)  # "/another/abs/path"
```
That is, concatenation of two absolute paths gives the latter one.

We could update this and other parts where we do a similar operation; would simplify the code (at the cost of worse readability!)

Comment (DavidMinarsch) [reply packages/fetchai/connections/p2p_libp2p/connection.py]: I fixed it now I think (data_dir is absolute, self.log_file is not as per check)

Comment (marcofavorito) [reply packages/fetchai/connections/p2p_libp2p/connection.py]: `data_dir` can be also a relative path wrt working directory. The concatenation would give the latter anyway if the latter is absolute

Comment (DavidMinarsch) [reply packages/fetchai/connections/p2p_libp2p/connection.py]: The latter cannot be absolute as per check!

**Review by solarw — APPROVED** [review]
LGTM

File: aea/multiplexer.py [hunk 2]
```diff
@@ -567,7 +567,11 @@ async def _send(self, envelope: Envelope) -> None:
         else:
             connection = self._id_to_connection[connection_id]
 
-        connection = cast(Connection, connection)
+        if connection is None:
```
Comment (DavidMinarsch) [inline aea/multiplexer.py:570]: This was forgotten in the previous PR which removed the stub connection (and therefore makes it possible to have no connection specified)

**Pull request merged**